### PR TITLE
CC-81: Added slide control to multi selector component

### PR DESCRIPTION
**Pull request by vishesh-ats**
resolves https://github.com/Components-care/backlog/issues/81

### CODE REVIEW

**Review by ankitp-atharva — DISMISSED** [review]

File: src/standalone/Selector/MultiSelect.tsx [hunk 1]
```diff
@@ -54,6 +54,18 @@ export interface MultiSelectProps<DataT extends MultiSelectorData>
 	subClasses?: {
 		baseSelector: BaseSelectorProps<BaseSelectorData>["classes"];
 	};
+	/**
+	 * Display switch control?
+	 */
+	displaySwitch?: boolean;
```
Comment (TQjona) [inline src/standalone/Selector/MultiSelect.tsx:60]: you don't need these, they are inherited from base selector props (see interface extends)

File: src/standalone/InlineSwitch/index.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	track: {
+		border: `1px solid ${theme.palette.grey[500]}`,
+		borderRadius: 16 / 2,
+		opacity: 1,
+		backgroundColor: theme.palette.common.white,
+	},
+	checked: {},
+}))(Switch);
+
+const InlineSwitch = (props: InlineSwitch) => {
```
Comment (TQjona) [inline src/standalone/InlineSwitch/index.tsx:73]: you should refactor MultiSelectWithTags component to use this instead

**Review by TQjona — CHANGES_REQUESTED** [review]
you can modify the update-cc script command to track a different branch so you can continue working on the main project without this getting merged into master

File: src/standalone/Selector/MultiSelect.tsx (hunk 1 above)
Comment (vishesh-ats) [reply src/standalone/Selector/MultiSelect.tsx]: > you don't need these, they are inherited from base selector props (see interface extends)
- @TQjona , Updated changes. Please Review.

File: src/standalone/InlineSwitch/index.tsx (hunk 2 above)
Comment (vishesh-ats) [reply src/standalone/InlineSwitch/index.tsx:79]: - @TQjona , Updated changes. Please Review.

File: src/standalone/Selector/BaseSelector.tsx [hunk 3]
```diff
@@ -75,7 +76,7 @@ export interface BaseSelectorProps<DataT extends BaseSelectorData>
 	 * Data load function
 	 * @param search The user search input
```
Comment (TQjona) [inline src/standalone/Selector/BaseSelector.tsx:76]: missing docs for param switchValue

File: src/standalone/InlineSwitch/index.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
+import React, { useCallback } from "react";
+import { Grid, Typography, withStyles, Theme, Switch } from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+
+export interface InlineSwitch {
+	/**
+	 * Value for switch position
+	 */
+	switchValue: boolean;
```
Comment (TQjona) [inline src/standalone/InlineSwitch/index.tsx:9]: rename to `value: boolean`

File: src/standalone/InlineSwitch/index.tsx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+	/**
+	 * Set value for switch position
+	 */
+	setSwitchValue: (checked: boolean) => void;
```
Comment (TQjona) [inline src/standalone/InlineSwitch/index.tsx:13]: rename to `onChange: (checked: boolean) => void`. also lacks docs for param checked

File: src/standalone/InlineSwitch/index.tsx [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+	/**
+	 * Display switch control?
+	 */
+	displaySwitch: boolean;
+	/**
+	 * Label for switch control (only used if displaySwitch is truthy)
+	 */
+	switchLabel?: React.ReactNode;
```
Comment (TQjona) [inline src/standalone/InlineSwitch/index.tsx:21]: rename to `label?: React.ReactNode`

File: src/standalone/InlineSwitch/index.tsx [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+	 */
+	switchValue: boolean;
+	/**
+	 * Set value for switch position
+	 */
+	setSwitchValue: (checked: boolean) => void;
+	/**
+	 * Display switch control?
+	 */
+	displaySwitch: boolean;
```
Comment (TQjona) [inline src/standalone/InlineSwitch/index.tsx:17]: rename to `visible: boolean`

File: src/standalone/InlineSwitch/index.tsx [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+	/**
+	 * Label for switch control (only used if displaySwitch is truthy)
+	 */
+	switchLabel?: React.ReactNode;
+	children?: React.ReactElement;
+	/**
+	 * Apply custom margin top class to set UI
+	 */
+	customMarginTop?: boolean;
```
Comment (TQjona) [inline src/standalone/InlineSwitch/index.tsx:26]: instead of putting custom styles in this control you should implement style overrides. for an example look at Loader component.

File: src/standalone/Selector/BaseSelector.tsx [hunk 9]
```diff
@@ -243,10 +256,15 @@ const BaseSelector = <DataT extends BaseSelectorData>(
 		noGroupLabel,
 		disableGroupSorting,
 		groupSorter,
+		switchLabel,
 	} = props;
 	const classes = useThemeStyles(
 		(props as unknown) as BaseSelectorProps<BaseSelectorData>
 	);
+	const defaultSwitchValue = props.displaySwitch
```
Comment (TQjona) [inline src/standalone/Selector/BaseSelector.tsx:264]: `const defaultSwitchValue = !!(props.displaySwitch && props.defaultSwitchValue)`

File: src/standalone/Selector/MultiSelectEntry.tsx [hunk 10]
```diff
@@ -10,7 +10,7 @@ import {
 import { SmallIconButton, SmallListItem, SmallListItemIcon } from "../Small";
 import { MultiSelectorData } from "./MultiSelect";
 import { ClassNameMap } from "@material-ui/styles/withStyles";
-import { RemoveIcon } from "../Icons";
```
Comment (TQjona) [inline src/standalone/Selector/MultiSelectEntry.tsx:13]: why did you change the icon?

**Review by TQjona — CHANGES_REQUESTED** [review]

File: src/standalone/Selector/MultiSelectEntry.tsx (hunk 10 above)
Comment (vishesh-ats) [reply src/standalone/Selector/MultiSelectEntry.tsx:13]: Same icon available in material UI icons. So, replaced with it.

File: src/standalone/Selector/BaseSelector.tsx (hunk 3 above)
Comment (vishesh-ats) [reply src/standalone/Selector/BaseSelector.tsx:76]: @TQjona , Changes updated. Please Review.

File: src/standalone/InlineSwitch/index.tsx (hunk 4 above)
Comment (vishesh-ats) [reply src/standalone/InlineSwitch/index.tsx]: @TQjona , Changes updated. Please Review.

File: src/standalone/InlineSwitch/index.tsx (hunk 5 above)
Comment (vishesh-ats) [reply src/standalone/InlineSwitch/index.tsx]: @TQjona , Changes updated. Please Review.

File: src/standalone/InlineSwitch/index.tsx (hunk 6 above)
Comment (vishesh-ats) [reply src/standalone/InlineSwitch/index.tsx]: @TQjona , Changes updated. Please Review.

File: src/standalone/InlineSwitch/index.tsx (hunk 7 above)
Comment (vishesh-ats) [reply src/standalone/InlineSwitch/index.tsx]: @TQjona , Changes updated. Please Review.

File: src/standalone/InlineSwitch/index.tsx (hunk 8 above)
Comment (vishesh-ats) [reply src/standalone/InlineSwitch/index.tsx]: @TQjona , Changes updated. Please Review.

File: src/standalone/Selector/BaseSelector.tsx (hunk 9 above)
Comment (vishesh-ats) [reply src/standalone/Selector/BaseSelector.tsx]: @TQjona , Changes updated. Please Review.

File: src/standalone/Selector/MultiSelectEntry.tsx (hunk 10 above)
Comment (TQjona) [reply src/standalone/Selector/MultiSelectEntry.tsx:13]: then delete the icon from components-care

File: src/standalone/Selector/BaseSelector.tsx [hunk 11]
```diff
@@ -243,10 +263,16 @@ const BaseSelector = <DataT extends BaseSelectorData>(
 		noGroupLabel,
 		disableGroupSorting,
 		groupSorter,
+		switchLabel,
 	} = props;
 	const classes = useThemeStyles(
 		(props as unknown) as BaseSelectorProps<BaseSelectorData>
 	);
+	const customClassesSwitch = useStyles();
```
Comment (TQjona) [inline src/standalone/Selector/BaseSelector.tsx:271]: any reason this is not part of the main styles?

```tsx
const customClassesSwitch = useMemo(() => ({
  switch: classes.switchStyles
}), [classes.switchStyles]);
```

**Review by TQjona — CHANGES_REQUESTED** [review]

File: src/standalone/Selector/MultiSelectEntry.tsx (hunk 10 above)
Comment (vishesh-ats) [reply src/standalone/Selector/MultiSelectEntry.tsx:13]: @TQjona , Removed from CC.

File: src/standalone/Selector/BaseSelector.tsx (hunk 11 above)
Comment (vishesh-ats) [reply src/standalone/Selector/BaseSelector.tsx:271]: @TQjona , It's just UI adjust for switch to fit in the layout properly for selector. So, I don't think it's consider as a main style.

**Review by ankitp-atharva — APPROVED** [review]

**Review by TQjona — APPROVED** [review]
